### ISSUE TITLE
Cast to TV

### ISSUE THREAD

**Issue by Bnyro** [libre-tube/LibreTube#799]
### Describe your suggested feature

Option to cast to a TV (e.g. Chromecast or FireTv Stick)

### Other details

I can't add this because I don't have the needed resources for it.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by FireMasterK** [comment]
https://developers.google.com/cast/docs/android_sender

It looks like you require a proprietary library :/ for doing it officially

But, it looks like we have a microG variant! https://github.com/microg/GmsCore/tree/master/play-services-cast-api

**Comment by lu4p** [comment]
I think it's okay to use the official api because it links against the library provided by the os i.e. if you don't have Google Play Services casting will not work for you.

**Comment by FireMasterK** [comment]
It's proprietary code, and would not be allowed in f-droid. I would rather suggest using the microg api, which is open source.

**Comment by the4anoni** [comment]
Remember that Chromecast uses Cast API, while other devices use pairing by code (YouTube.com/activate). Second option provides support for much more devices (Tizen, WebOS TVs, SmartTubeNext).

SmartTubeNext is open source alternative client independent on gapps that works on any Android device meant for TV (doesn't matter is this official Android TV, non certified Chinese Android Box, Fire TV, etc.).

I think pairing by code should be higher on priority list that casting, also it looks like easier to implement.

**Comment by vighd** [comment]
This can be achieved easily without any development with the current version. If you want to cast a video, just press the "Open..." button right beside the Download button and cast the video with google photos video player.

**Comment by the4anoni** [comment]
> This can be achieved easily without any development with the current version. If you want to cast a video, just press the "Open..." button right beside the Download button and cast the video with google photos video player.

I don't have installed Google Photos, also It would be nice if we could have button in app for that.

**Comment by vighd** [comment]
> > This can be achieved easily without any development with the current version. If you want to cast a video, just press the "Open..." button right beside the Download button and cast the video with google photos video player.
> 
> I don't have installed Google Photos, also It would be nice if we could have button in app for that.

Yeah, that would be nice too, I just want to share an alternative way.

**Comment by xIUPITERx** [comment]
> This can be achieved easily without any development with the current version. If you want to cast a video, just press the "Open..." button right beside the Download button and cast the video with google photos video player.

The button was removed in the last update.

**Comment by Bnyro** [comment]
It might have a comeback with #2292 though, didn't decide it yet.

**Comment by youdontneedtoknow22** [comment]
Skytube solved this problem with having 2 Versions, one is a FOSS version on F-Droid without the cast feature, and an extra version hosted on Github with Chromecast.

https://github.com/SkyTubeTeam/SkyTube/issues/22

**Comment by xIUPITERx** [comment]
Maybe I'm an optimist, but I just hope that Matter TV will catch on and that it will be open source. but it won't be released until the end of this year at the earliest.

**Comment by the4anoni** [comment]
> 

https://imgs.xkcd.com/comics/standards.png

**Comment by jaredsburrows** [comment]
You can with NewPipe + VLC. See https://github.com/TeamNewPipe/NewPipe/issues/668#issuecomment-959954722.

Possible we can do something like that here?
